src/ILCompiler.Compiler/src/Compiler/CompilerGeneratedMetadataManager.cs

```diff
@@ -110,6 +114,52 @@ protected override MetadataCategory GetMetadataCategory(TypeDesc type)
             // Generate metadata blob
             var writer = new MetadataWriter();
             writer.ScopeDefinitions.AddRange(transformed.Scopes);
+
+            // Generate entries in the blob for methods that will be necessary for stack trace purposes.
```

Comment:
This method is getting rather large. Would it make sense to split into a set of smaller ones?

Reply:
I'm not convinced splitting it up would result in more readable code. There's a bunch of state that will need to be shared between the smaller methods. The methods will get shorter, but tracking the state overall will get harder.